Automattic/woocommerce-payments
# Update Subscriptions with WooPayments eligibility as we move to deprecate this functionality

> [!NOTE]  
> This PR reintroduces the changes made in #7117 that were reverted in #7194 when it was decided that deprecating the WCPay Subscriptions feature would come in the next release. 

Fixes #6510 

#### Changes proposed in this Pull Request









As part of the [Subscriptions Primary Offering](https://github.com/Automattic/woocommerce-payments/milestone/34) project, we need to deprecate the existing WCPay Subscriptions feature so that merchants can opt into the new subscriptions setting.

In this PR we are updating the eligibility of the WCPay Subscriptions so that we:
 - Do not load the **Advanced > WCPay Subscription** setting on new sites created after the release of the version of WC Pay containing these changes.
 - No longer load subscription functionality on existing sites that have no active Stripe Billing Subscriptions or published subscription products.

To meet these new requirements I've changed `is_wcpay_subscriptions_eligible()` to only return true when:

1. Store located in the US
2. Doesn't have the WC Subscriptions extension active
3. Has either one of the following:
   1. the store has existing WCPay Subscriptions or,
   2. has published subscription products

#### Testing instructions





Testing this PR requires toggling the following:
 - Deactivate/activate WooCommerce Subscriptions
 - Update the store's base country to US and non-US
 - Has no published subscription products
 - Has no WCPay subscriptions (with `_wcpay_subscription_id` metadata)

With that information, here are the testing instructions to toggle through most of the new eligibility cases:

1. Setup the state of your store so that you have 1 subscription product, 1 WCPay Subscription, WC Subscriptions deactivated, and the store is based in the US
2. Confirm the **Enable Subscriptions with WooPayments** setting is found in **Payments > Settings > Advanced**
3. Activate Subscriptions, confirm the setting is gone
4. Deactivate Subscriptions to bring it back
5. Change the stores base country to a non-US country, confirm the WCPay Subscriptions setting is gone
6. Change the country back to the US
7. Set your subscriptions product to private and modify the `_wcpay_subscription_id` to something like `_wcpay_subscription_id_test`
8. Confirm the setting is no longer shown
9. Set the subscription product back to published, confirm the setting is back.
10. Put the product back to private and change the metadata on the subscription to bring back the `_wcpay_subscription_id` meta.
11. Confirm the WCPay subscription setting is loading.

With these instructions I have tested the following cases:

- **Setting doesn't show if any of the following conditions are met:**
   - [x] Has WC Subscriptions active
   - [x] Non-US based store
   - [x] Doesn't have existing subscriptions products
   - [x] Doesn't have any subscriptions with `__wcpay_subscriptions_id` meta (including pending, canceled, on-hold etc.)
- **Settings only shows when:**
   - [x] No WC Subscriptions active, country set to US, no subscriptions but has existing products
   - [x] No WC Subscriptions active, country set to US, has existing subscriptions 

-------------------


**Post merge**
